src/quark-core/src/QuarkScript.sol

```diff
@@ -79,7 +96,11 @@ abstract contract QuarkScript {
     }
 
     function read(bytes32 key) internal view returns (bytes32) {
-        return QuarkWallet(payable(address(this))).stateManager().read(key);
+        bytes32 value;
```

Comment:
Almost wonder if we should make a `mapping (bytes32 => bytes32) sharedStorage` in the Wallet and then just use that. It's maybe a little inferior (e.g. it's based on slot # so scripts needs to understand that), though it adds a small benefit that it's much clearer that it's shared. Could maybe even be `mapping (bytes32 => mapping(bytes32 => bytes32) nonceSharedStorage`

Reply:
At the least, I wonder if these scripts should take in or depend on the `nonce` value (e.g. `tload`)? That could help make them _more isolated_, though not fully.